<launch>
  <!-- Launch file to run bag file -->
  <arg name="bag_file_path" default=""/>

  <!-- Camera image left topic -->
  <arg name="left_camera_image_topic" default="/zed_lateral/zed_lateral/left/image_rect_color"/>
  <!-- Camera image right topic -->
  <arg name="right_camera_image_topic" default="/zed_lateral/zed_lateral/right/image_rect_color"/>
  <!-- Camera info left topic -->
  <arg name="left_camera_info_topic" default="/zed_lateral/zed_lateral/left/camera_info"/>
  <!-- Camera info right topic -->
  <arg name="right_camera_info_topic" default="/zed_lateral/zed_lateral/right/camera_info"/>

  <!-- Play the rosbag and remap camera info topics -->
  <node pkg="rosbag" type="play" name="rosbag_play" args="$(arg bag_file_path) $(arg left_camera_info_topic):=/stereo/left/camera_info $(arg right_camera_info_topic):=/stereo/right/camera_info" output="screen"/>

  <!-- Remap left camera image topic -->
  <node pkg="image_transport" type="republish" name="remap_left_image" output="screen" args="compressed in:=$(arg left_camera_image_topic) raw out:=/stereo/left/image_raw" />
  <!-- Remap right camera image topic -->
  <node pkg="image_transport" type="republish" name="remap_right_image" output="screen" args="compressed in:=$(arg right_camera_image_topic) raw out:=/stereo/right/image_raw" />
</launch>
